src/Pages/Signup.js

```diff
@@ -0,0 +1,116 @@
+import axios from 'axios'
+import {Link,useNavigate} from 'react-router-dom'
+
+export function Signup() {
+    const navigate = useNavigate()
+
+    const [textInputState,textInputDispatcher] = useReducer(inputDispatcher,{username:"",password:"",rePassword:"",answer:""})
+    
+    const [visibilityErr,setErrVisibility] = useState("hidden")
+    const [Alertclass,setAlertClass] = useState("alert")
+    const [faClass,setFaClass] = useState("fa-check-circle-o")
+    const [content,setContent] = useState("")
+
+    const [signupContent,setSignupContent] = useState("Sign up")
+    async function userSignup(){
+        try{
+        if(textInputState.username !== "" && textInputState.password !== "" && textInputState.answer !== ""){
+            if(textInputState.password === textInputState.rePassword){
+                setSignupContent("Signing up...")
+            const response_signup = await axios.post("https://KnightStream.ishanjirety.repl.co/api/signup",textInputState)
+            if(response_signup.data.status===200){
+                setContent("User created successfully")
+                setAlertClass("alert animate")
+                setErrVisibility("visible")
+                setFaClass("fa-check-circle-o")
+                setSignupContent("Sign up")
+                navigate('/login')
+            }
+            else{
+                setContent("Could not create user")
+                setAlertClass("alert danger animate")
+                setErrVisibility("visible")
+                setFaClass("fa-exclamation")
+            }
+            }else{
+                setErrVisibility("visible")
+                setContent("Password does not match")
+                setAlertClass("alert danger animate")
+                setFaClass("fa-exclamation")
+            }
+        }
+        else{
+                setErrVisibility("visible")
+                setContent("Fields cannot be empty")
+                setAlertClass("alert danger animate")
+                setFaClass("fa-exclamation")
+        }
+        }catch(e){
+            setContent("Uh no! 500 Internal server error")
+            setAlertClass("alert danger animate")
+            setErrVisibility("visible")
+            setFaClass("fa-exclamation")
+            setSignupContent("Sign up")
+        }
+    } 
+
+    useEffect(()=>{setTimeout(()=>{
+            setAlertClass("alert fade")
+            setErrVisibility("hidden")
+        },4000)
+    },[visibilityErr])
+    return (
+        <div className="login">
+            <div className="login-logo">
+                <span className="login-underline">
+                    <p className="login-border">KnightStream</p>
+                </span>
+            </div>  
+            <section className="login-card signup-card">
+                <p className="login-greeting">WELCOME</p>
+                <p className="login-heading">Create Account</p>
+                
+                <div className="input-fields">   
```

Comment:
You can wrap this whole thing in `<form>` element. It semantically correct and also improves accessibility, especially while using keyboard.